core/services/ocr2/plugins/ocr2vrf/blockhashes/fixed_blockhash_provider.go

```diff
@@ -41,59 +49,77 @@ func (b *fixedBlockhashProvider) OnchainVerifiableBlocks(
 	if err != nil {
 		return 0, nil, errors.Wrap(err, "current height")
 	}
-	fromBlock := toBlock - b.lookbackBlocks
-	var reqs []rpc.BatchElem
 
+	fromBlock := uint64(0)
+	if toBlock > b.lookbackBlocks {
+		fromBlock = toBlock - b.lookbackBlocks
+	}
+
+	var blockHeights []uint64
 	for i := fromBlock; i <= toBlock; i++ {
-		req := rpc.BatchElem{
-			Method: "eth_getBlockByNumber",
-			Args:   []interface{}{hexutil.EncodeBig(big.NewInt(int64(i))), false},
-			Result: &evmtypes.Head{},
+		blockHeights = append(blockHeights, i)
+	}
+
+	blockhashes := make([]common.Hash, len(blockHeights))
+
+	heads, err := b.lp.GetBlocks(blockHeights, pg.WithParentCtx(ctx))
+	if err != nil {
+		b.logger.Warnw("error in get blocks", "err", err)
```

Comment:
Could make this a bit more informative: `"Failed to GetBlocks from log poller, falling back to RPC"`